particula/aerosol_dynamics/parcel.py

```diff
@@ -48,43 +38,68 @@ def name(self) -> str:
     def add_particle(self, particle_object):
         """Adds a particle to the parcel.
         Parameters:
-            particle_data (list): The radius of the particle.
+            particle_object (object)
         """
-        self._particle_data.append(particle_object)
+        self._particle_data = np.append(self._particle_data, particle_object)
 
     def create_and_add_particle(
-        self, name: str, radius, density=1e3, charge=0
+        self, name: str, radius, density=1e3 * u.kg / u.m**3,
+        charge=0 * u.dimensionless
     ):
         """creats and then Adds a particle to the parcel.
         Parameters:
-            particle_data (list): The radius of the particle.
+            name    (str)   [no units],
+            radius  (float) [m]
+        Optional:
+            density (float) [kg/m**3]       default = 1e3 [kg/m**3],
+            charge  (int)   [dimensionless] default = 0 [dimensionless]
         """
-        self._particle_data.append(
+
+        try:  # maybe move to particle class
+            radius.check('m')
+        except AttributeError:
+            print('Please add units to radius. E.g., radius = 150 * u.nm]')
+
+        self.add_particle(
             particle.Particle(name, radius, density, charge)
         )
 
     def create_and_add_list_of_particle(
-        self, name_of_particles, radius_of_particles
+        self, name_of_particles, radius_of_particles,
+        density_of_particles=None, charge_of_particles=None
     ):
-        """Adds a list of particles to the parcel based on size only.
+        """Adds a list of particles to the parcel based on size only
+            or delcaring denisty and charge.
         Parameters:
-            name_of_particles (list): The name of the particles.
-            radius_of_particles (list): The radius of the particle.
-            TODO: add charge and mass as passible parameters.
+            name_of_particles   (str)   [no units],
+            radius_of_particles (list)  [m]
+        Optional:
+            density_of_particles (list) [kg/m**3],
+            charge_of_particles  (list) [dimensionless]
```

Comment:
How do you feel about `charge_of_particles` vs `charges_of_particles` to describe lists of things? I think it may be more intuitive to use the plural for both?

Reply:
yeah, and same for radi and densities. Also it make sense that you can expect more than one value in return